frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -631,16 +642,24 @@ pub mod pallet {
 			}
 		}
 
-		fn offchain_worker(n: T::BlockNumber) {
-			// We only run the OCW in the first block of the unsigned phase.
-			if Self::current_phase().is_unsigned_open_at(n) {
-				match Self::try_acquire_offchain_lock(n) {
-					Ok(_) => {
-						let outcome = Self::mine_check_and_submit().map_err(ElectionError::from);
-						log!(info, "mine_check_and_submit execution done: {:?}", outcome);
+		fn offchain_worker(now: T::BlockNumber) {
+			match Self::current_phase() {
+				Phase::Unsigned((true, opened)) if opened == now => {
+					// mine a new solution, cache it, and attempt to submit it
+					let initial_output = Self::try_acquire_offchain_lock(now)
```

Comment:
what do you think about making either a closure or macro that guards the execution of an arbitrary function with `try_acquire_offchain_lock `, and prints trace logs when the lock failed to be acquired? I wanted to recommend adding this log, but we have to clutter the code now and duplicate it in two places.